### ISSUE TITLE
[Bug] "enableTypeIgnoreComments" is an unknown diagnostic rule for pyright comment

### ISSUE THREAD

**Issue by F-park** [microsoft/pyright#7078]
## Bug Description

There is `enableTypeIgnoreComments` in [diagnostic-rule-defaults](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#diagnostic-rule-defaults), but it no work.

## Code

```py
# pyright: enableTypeIgnoreComments=true

```

## pyright version

v1.1.347

**Comment by erictraut** [comment]
I just verified that it's working fine. I suspect you're using the feature incorrectly.

Type ignore comments (of the form `# type: ignore`) are enabled by default. If you would like to disable them — effectively telling pyright to ignore their presence, you can set `enableTypeIgnoreComments` to false in your pyright configuration.

Normally the following type ignore comment would suppress the type error on this line.
```python
x: int = "" # type: ignore
```

Now create a `pyrightConfig.json` file that contains the following:

```json
{
    "enableTypeIgnoreComments": false
}
```

You will now see that the type error is reported despite the fact that you have a `# type: ignore` comment on that line.

**Comment by erictraut** [comment]
Yes, `enableTypeIgnoreComments` is intended to be used in the config file and will not work if used in a file-level comment. It affects a part of the type analyzer that is applied after all other processing across all files.
